src/sagemaker/multidatamodel.py

```diff
@@ -0,0 +1,211 @@
+        self.image = image
+        self.role = role
+        self.predictor_cls = predictor_cls
+
+        # Copy the values from the model if these attributes are not provided in the constructor
+        if self.model:
+            self.image = self.image or self.model.image
+            self.role = self.role or self.model.role
+            self.predictor_cls = self.predictor_cls or self.model.predictor_cls
+
+        super(MultiDataModel, self).__init__(
+            self.model_data_prefix,
+            self.image,
+            self.role,
+            name=self.model_name,
+            predictor_cls=self.predictor_cls,
+            **kwargs
+        )
+
+        if not self.sagemaker_session:
+            self.sagemaker_session = sagemaker.session.Session()
+        self.s3_client = self.sagemaker_session.boto_session.client("s3")
+
+        # Create the S3 prefix path to ensure model deployment succeeds
+        self._create_s3_model_data_path()
+
+    @property
+    def model_data_prefix(self):
+        """Placeholder docstring"""
+        return self._model_data_prefix
+
+    @model_data_prefix.setter
+    def model_data_prefix(self, model_data_prefix):
+        """
+        Args:
+            model_data_prefix:
+        """
+        # Validate path
+        if not (model_data_prefix.startswith("s3://") and model_data_prefix.endswith("/")):
+            raise ValueError(
+                'Expecting S3 model prefix beginning with "s3://" '
+                'and ending in "/". Received: "{}"'.format(model_data_prefix)
+            )
+        self._model_data_prefix = model_data_prefix
+
+    def _create_s3_model_data_path(self):
+        """
+        Create the S3 prefix path to ensure model deployment succeeds.
+        If this path does not exist, calls to CreateModel API fails
+        """
+        bucket, model_data_path = self._parse_s3_uri(self.model_data_prefix)
+        self.s3_client.put_object(Bucket=bucket, Key=os.path.join(model_data_path, "/"))
+
+    def prepare_container_def(self, instance_type, accelerator_type=None):
+        """Return a container definition set with MultiModel mode and
+        model data and all other parameters from the original model pass
+        to the MultiDataModel constructor
+
+        Subclasses can override this to provide custom container definitions
+        for deployment to a specific instance type. Called by ``deploy()``.
+
+        Returns:
+            dict[str, str]: A complete container definition object usable with the CreateModel API
+        """
+        # Copy the trained model's image and environment variables if they exist.
+        # Models trained with FrameworkEstimator set framework specific environment variables
+        # which need to be copied over
+        if self.model:
+            container_definition = self.model.prepare_container_def(instance_type, accelerator_type)
+            self.image = self.image or container_definition["Image"]
```

Comment:
This was handling the case when self.image is None, in which case it would be updated with the value of `container_definition["Image"]`. But I have updated the logic now to use `self.model` params when available and ignore local params per our discussion